packages/core/content-releases/shared/types.ts

```diff
@@ -24,3 +24,26 @@ export interface Entity {
   createdAt: string;
   updatedAt: string;
 }
+
+/**
+ * Utility type to enforce mutually exclusive properties.
+ *
+ * This type ensures that either properties of type {@link T} or properties of type {@link U} are present,
+ * but never both at the same time. It is useful for defining states where you want to
+ * have exactly one of two possible sets of properties.
+ *
+ * @template T - The first set of properties.
+ * @template U - The second set of properties.
+ *
+ * @example
+ * // Define a type where you either have data or an error, but not both:
+ * type Response = OneOf<
+ *   { data: Data },
+ *   { error: ApplicationError | ValidationError }
+ * >;
+ *
+ * // Is equivalent to:
+ * type Response = { data: Data, error: never } | { data: never, error: ApplicationError | ValidationError };
+ *
+ */
+export type OneOf<T, U> = (T & { [K in keyof U]?: never }) | (U & { [K in keyof T]?: never });
```

Comment:
@Convly Do you think I could fit a utility like this into our types package? This is something I could start using a lot in our type contract files

Reply:
It could actually replace the usage we're having of [XOR](https://github.com/strapi/strapi/blob/v5/main/packages/core/types/src/utils/index.ts#L99-L126) in a lot of places.

Go for it :slightly_smiling_face: